vanilla/vanilla
# Make OAuth2 Plugin OIDC compliant

ref: https://github.com/vanilla/support/issues/464

This PR will add several features to our OAuth2 core class to make it more compliant. In this PR we will:

 - Add the default data keys for translating OIDC profile data to Vanilla's User values.
 - Allow admins to choose what Prompt type they will send in the authorization request (`none`, `consent`, `consent and login`).
 - Make configurable by the CSMs the `response_type` being sent. Presently it is hard-coded as `code`.

### Testing

 - Start with a fresh OAuth2 set up. Delete your OAuth2 entry in GDN_UserAutheticationProvider.
 - Turn on the OAuth2 plugin.
    - Check that the default values for Name is nickname and UserID is sub
    - Check that there is a Prompt field.
 - Configure the plugin to point to a OAuth2 application.
 - Set the prompt to `none`.
   - check that, when you log out, if you log in again, you are not prompted again to login at the authentication provider.
   - test the other prompt settings.
